src/main/java/seedu/address/storage/JsonAdaptedPropertyForSale.java

```diff
@@ -0,0 +1,92 @@
+import seedu.address.model.property.PropertyForSale;
+
+/**
+ * Jackson-friendly version of {@link PropertyForSale}.
+ */
+class JsonAdaptedPropertyForSale {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "PropertyForSale's %s field is missing!";
+    private final String address;
+    private final String town;
+    private final String propertyType;
+    private final double size;
+    private final int numberOfBedrooms;
+    private final int numberOfBathrooms;
+    private final double price;
+    private final Optional<String> remark;
+    private final String listingDate; // String representation of the LocalDate
+
+    /**
+     * Constructs a {@code JsonAdaptedPropertyForSale} with the given property details.
+     */
+    @JsonCreator
+    public JsonAdaptedPropertyForSale(@JsonProperty("address") String address,
+                                      @JsonProperty("town") String town,
+                                      @JsonProperty("propertyType") String propertyType,
+                                      @JsonProperty("size") double size,
+                                      @JsonProperty("numberOfBedrooms") int numberOfBedrooms,
+                                      @JsonProperty("numberOfBathrooms") int numberOfBathrooms,
+                                      @JsonProperty("price") double price,
+                                      @JsonProperty("remark") String remark,
+                                      @JsonProperty("listingDate") String listingDate) {
+        this.address = address;
+        this.town = town;
+        this.propertyType = propertyType;
+        this.size = size;
+        this.numberOfBedrooms = numberOfBedrooms;
+        this.numberOfBathrooms = numberOfBathrooms;
+        this.price = price;
+        this.remark = Optional.ofNullable(remark);
+        this.listingDate = listingDate;
+    }
+
+    /**
+     * Converts a given {@code PropertyForSale} into this class for Jackson use.
+     */
+    public JsonAdaptedPropertyForSale(PropertyForSale source) {
+        address = source.getAddress();
+        town = source.getTown();
+        propertyType = source.getPropertyType();
+        size = source.getSize();
+        numberOfBedrooms = source.getNumberOfBedrooms();
+        numberOfBathrooms = source.getNumberOfBathrooms();
+        price = source.getPrice();
+        remark = source.getRemark(); // Convert Optional<String> to String
+        listingDate = source.getListingDate().toString();
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted property object into the model's {@code PropertyForSale} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted property.
+     */
+    public PropertyForSale toModelType() throws IllegalValueException {
+        if (address == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "address"));
```

Comment:
Same comment as above (second argument in String.format needs to follow the Name.class.getSimple__() format, similar to JsonAdaptedPerson.java (for all 3 conditionals)) for "address", "town", "propertyType" and "listingDate"